core/cmd/shell.go

```diff
@@ -60,11 +60,18 @@ var (
 	grpcOpts        loop.GRPCOpts
 )
 
-func initGlobals(cfg config.Prometheus) {
-	// Avoid double initializations.
+func initGlobals(cfgProm config.Prometheus, cfgTracing config.Tracing) {
+	// Avoid double initializations, but does not prevent relay methods from being called multiple times.
 	initGlobalsOnce.Do(func() {
-		prometheus = ginprom.New(ginprom.Namespace("service"), ginprom.Token(cfg.AuthToken()))
-		grpcOpts = loop.SetupTelemetry(nil) // default prometheus.Registerer
+		prometheus = ginprom.New(ginprom.Namespace("service"), ginprom.Token(cfgProm.AuthToken()))
+		grpcOpts = loop.NewGRPCOpts(nil) // default prometheus.Registerer
+		err := loop.SetupTracing(loop.TracingConfig{
+			Enabled:         cfgTracing.Enabled(),
+			CollectorTarget: cfgTracing.CollectorTarget(),
+			NodeAttributes:  cfgTracing.Attributes(),
+			SamplingRatio:   cfgTracing.SamplingRatio(),
+		})
+		fmt.Printf("Failed tracing setup %v", err)
```

Comment:
Let's log this through a `logger.Logger` so it is not overlooked as malformed JSON in the log stream. Could pass it in, or maybe simpler to return an error from here to log in `NewApplication`

Reply:
I think something is missing from my mental model here - why would a fmt statement here result in malformed JSON in the log stream? 

From a readability perspective I don't think logging the error in NewApplication is better. It looks like this:

```
func (n ChainlinkAppFactory) NewApplication(ctx context.Context, cfg chainlink.GeneralConfig, appLggr logger.Logger, db *sqlx.DB) (app chainlink.Application, err error) {
	err = initGlobals(cfg.Prometheus(), cfg.Tracing())
	if err != nil {
		appLggr.Errorf("Failed to initialize globals: %v", err)
	}
```

Though when compared to drilling a logger just to log the error, I'm inclined to leave it in NewApplication.

Reply:
> why would a fmt statement here result in malformed JSON in the log stream?

Bypassing the logger with `fmt.Print` generates raw output, and it would be considered malformed when JSON is configured. Grafana filter queries commonly discard malformed lines.